File: app/src/main/java/br/com/jonathanzanella/myexpenses/helpers/DateHelper.java
package br.com.jonathanzanella.myexpenses.helpers;

import org.joda.time.DateTime;

public final class DateHelper {
	private DateHelper() {}

	public static DateTime firstDayOfMonth(DateTime date) {
		return firstMillisOfDay(date.dayOfMonth().withMinimumValue());
	}

	public static DateTime firstMillisOfDay(DateTime date) {
		return date.withMillisOfDay(0);
	}

	public static DateTime lastDayOfMonth(DateTime date) {
		return lastMillisOfDay(date.dayOfMonth().withMaximumValue());
	}

	public static DateTime lastMillisOfDay(DateTime date) {
		return date.withTime(23, 59, 59, 999);
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/helpers/DateHelper.kt
package br.com.jonathanzanella.myexpenses.helpers

import org.joda.time.DateTime

object DateHelper {
    private val MAX_HOURS_OF_DAY = 23
    private val MAX_MINUTES_OF_HOUR = 59
    private val MAX_SECONDS_OF_MINUTE = 59
    private val MAX_MILLISECONDS_OF_SECOND = 999

    @JvmStatic fun firstDayOfMonth(date: DateTime): DateTime {
        return firstMillisOfDay(date.dayOfMonth().withMinimumValue())
    }

    @JvmStatic fun firstMillisOfDay(date: DateTime): DateTime {
        return date.withMillisOfDay(0)
    }

    @JvmStatic fun lastDayOfMonth(date: DateTime): DateTime {
        return lastMillisOfDay(date.dayOfMonth().withMaximumValue())
    }

    @JvmStatic fun lastMillisOfDay(date: DateTime): DateTime {
        return date.withTime(MAX_HOURS_OF_DAY, MAX_MINUTES_OF_HOUR, MAX_SECONDS_OF_MINUTE, MAX_MILLISECONDS_OF_SECOND)
    }
}